app/helpers/file-system.js

```diff
@@ -135,6 +135,24 @@ module.exports = {
     });
   },
 
+  isProcessing: function(fileName) {
+    return global.PROCESSING_LIST.indexOf(fileName) !== -1;
+  },
+
+  addToProcessingList: function(fileName) {
+    let index = global.PROCESSING_LIST.indexOf(fileName);
+    if (index === -1) {
+      global.PROCESSING_LIST.push(fileName);
+    }
+  },
+
+  removeFromProcessingList: function(fileName) {
+    let index = global.PROCESSING_LIST.indexOf(fileName);
+    if (index !== -1) {
+      global.PROCESSING_LIST.splice(index, 1);
+    }
+  },
+
```

Comment:
Looks like a good place to use [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set).